apps/gitness/src/framework/context/ExitConfirmContext.tsx

```diff
@@ -0,0 +1,51 @@
+import React, { createContext, useCallback, useMemo, useState } from 'react'
+import { ExitConfirmDialog } from '../../components/ExitConfirmDialog'
+
+export interface ExitConfirmOptions {
+  title?: string
+  subtitle?: string
+  confirmText?: string
+  cancelText?: string
+  onConfirm?: () => void
+  onCancel?: () => void
```

Comment:
I think these callbacks should be required. the display strings can have default values to make them optional

Reply:
Updated. Callbacks are mandatory.